Cierva cerinea 5

josevilchez247 opened the pull request:
* [x] ¿Se han establecido criterios previos de elección de la imagen base, así
como los criterios de búsqueda que han conducido a las imágenes que se están
considerando?
* [x] ¿Se ha tenido en cuenta que la imagen es para testear, y que por tanto tiene una función específica, tal como testear cómo va en la última versión publicada del lenguage?
* [x] ¿Se ha tenido en cuenta que la versión de la imagen base sea la última?
* [x] ¿Se han tenido en cuenta criterios reales para la elección, y no criterios irrelevantes como cuanto tiempo se tarda en construirla? (Se va a construir una vez, se va a usar cientos o miles de veces)
* [x] ¿Se han seguido las buenas prácticas en las órdenes que construyen
   la imagen? ¿Especialmente las relativas a usuarios no privilegiados y demás?
* [x] ¿La imagen incluye sólo y exclusivamente la infraestructura
   necesaria para pasar los tests? Por ejemplo, si se ha elegido la imagen
   "oficial" (lo que hay que hacer sólo en casos muy justificados) ¿se ha
   comprobado que lo que se instale sea sólo y exclusivamente lo necesario para
   nuestro proyecto?
* [x] ¿Se han documentado y enlazado los commits a las imágenes que se
   han testeado?
* [x] ¿Está establecido correctamente como `WORKDIR` el que se va a usar en los
   tests que se van a lanzar?
* [x] ¿Se usa algún directorio adecuado para copiar contenidos del repositorio
en vez del directorio de trabajo?
* [x] ¿Se han establecido claramente los criterios de búsqueda y de aceptación de
   la imagen base?
* [x] ¿Tienes claro cuales son las condiciones en las que tiene que actualizarse
la imagen en DockerHub y has configurado el workflow de acuerdo con ello? En
particular, ¿sabes qué condiciones tienen que darse para que se actualice y qué
condiciones *no* deben darse?
* [x] Si se usan GitHub actions para su publicación, ¿se han considerado cuidadosamene los criterios (ramas, paths) para que se lancen?

josevilchez247 committed:
Documentación de los contenedores docker

Avanza #29

josevilchez247 committed:
Actualizo readme

Avanza #28

josevilchez247 committed:
Añado Dockerfile

Avanza #29

josevilchez247 committed:
Correción Dockerfile

Avanza #29

josevilchez247 committed:
Añado el script

Avanza #29

josevilchez247 committed:
Modifico scripts

Avanza #29

josevilchez247 committed:
Creo Github Action

Avanza #30

josevilchez247 committed:
Añado github action

Avanza #30

josevilchez247 committed:
Añado nueva lógica de negocio

Avanza #31, implemento una matriz de adyacencia y el algoritmo de Dijkstra

josevilchez247 committed:
Cambio credenciales

josevilchez247 committed:
Update build.yml

josevilchez247 committed:
Update build.yml

josevilchez247 committed:
Update build.yml

josevilchez247 committed:
Update build.yml

josevilchez247 committed:
Actualizo Dockerfile

Avanza #29

josevilchez247 committed:
Actualizo DockerFile

Avanza #29

josevilchez247 committed:
Actualizo DockerFile

Avanza #29

josevilchez247 committed:
Actualizo DockerFile

Avanza #30

josevilchez247 committed:
Actualizo dependencias

#30

josevilchez247 committed:
Actualizo DockerFile

Avanza #30

josevilchez247 committed:
Actualizo DockerFile

josevilchez247 committed:
Actualizo Dependencias

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Actualizo dockerfile

#30 avanza

josevilchez247 committed:
Actualizo dockerfile

Avanaza #30

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Error de sintaxis

Avanza #30

josevilchez247 committed:
Actualizo Dockerfile

josevilchez247 committed:
Actualizo Dockerfile

#30 avanza

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Error sintaxis

josevilchez247 committed:
Actualizo Dockerfile

#30 avanza

josevilchez247 committed:
Actualiza Dockerfile

Avanza #30

josevilchez247 committed:
Actualizo Dockerfile

Avanzo #30

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Actualizo Dockerfile

josevilchez247 committed:
Actualizo dockerfile

Avanza #30

josevilchez247 committed:
Actualizo Dockerfile

#30 avanza

josevilchez247 committed:
Actualizo dependencias

#29

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Actualizo dockerfile

#30 avanza

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Actualizo Dockerfile

avanza #30

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update pyproject.toml

josevilchez247 committed:
Update pyproject.toml

josevilchez247 committed:
Actualizo dockerfile

Avanza #30

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Actualizo Dcokerfile

#30 avanza

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Añado poetry.lock

Avanza #25

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Actualizo Dockerfile

josevilchez247 committed:
Update Dockerfile

#30 avanza

josevilchez247 committed:
Update poetry.lock

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Actualizo Dockerfile

Avanza #30

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update build.yml

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update pyproject.toml

josevilchez247 committed:
Update Dockerfile

josevilchez247 committed:
Update poetry.lock

josevilchez247 committed:
Add files via upload

josevilchez247 committed:
Actualizo Readme

#28 Avanza

josevilchez247 committed:
Actualizo Dependencias

#30 avanza

josevilchez247 committed:
Actualizo dockerfile

#30 avanza

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,29 @@
+WORKDIR /app/test
+
+# Movemos el fichero de depencencias y el fuente del gestor de tareas al directorio de trabajo.
+COPY poetry.lock pyproject.toml task.py /app/test/
```
Sólo puedes copiar al contenedor lo que necesites para construirlo, y *nunca* en /app/test que se borrará cuando se monte el directorio externo.

josevilchez247 replied:
De acuerdo lo cambio

JJ commented on docs/docker.md:
```diff
@@ -0,0 +1,17 @@
+las necesidades del proyecto. Como este proyecto es algo académico y solamente se va a utilizar para ejecutar los test,no necesitamos una imagen que pese demasiado y que tenga una 
+gran cantidad de herramientas ya instaladas.
+
+Las opciones que he estudiado han sido [Python](https://hub.docker.com/_/python) y [Alpine](https://hub.docker.com/_/alpine), basándome en lo dicho anteriormente he decido usar
```
No has hecho ninguna comparación de contenedores y te has limitado a mirar los contenedores oficiales. No has hecho ninguna comparación específica para tu problema.

josevilchez247 replied:
Probé con estos, pero es verdad que no he puesto nada comparándolos y justificar la elección de python, leí ambas documentaciones y empecé probando python:3-9 ya que era la última versión pero la imagen pesaba mucho y entonces dije voy a probar con alpine y más o menos estaba en las mismas, seguí investigando y vi las versiones slim probe la python:3.9-slim y esa ya si bajo considerablemente el peso de la imagen, así que me decidí por esa.

JJ commented on task.py:
```diff
@@ -0,0 +1,24 @@
+    Lanzará todos los tests definidos en el directorio tests
+    """
+
+    print("Lanzando tests...")
```
Esto no hace nada, sólo imprime cosas. No lanza los tests.

josevilchez247 replied:
Si profesor este fue un fichero que puse ya en las últimas pruebas que hice lo borro, inicialmente no estaba

JJ commented:
Ninguno de los commits se ha hecho a partir de un issue.

josevilchez247 commented:
Voy a modificar varias cosas que están de la última prueba que hice, y voy a volver a ponerlo como lo envié las primeras veces, al principio todos los commit estaban con sus correspondientes issue pero a medida que me iba dando problemas empecé a mandar commit y no los puse con su issue correspondiente lo dicho como voy a volver a revisar todos los ficheros a partir de eso todos los commit que haga irán con los issues que avancen.

josevilchez247 committed:
Actualizo el Readme

Avanza #28

josevilchez247 committed:
Actualizo Dockerfile

Avanza #29 y #30

josevilchez247 committed:
Actualizo pyproject.toml

Avanza #29

josevilchez247 committed:
Actualizo build.yaml

Avanza #30

josevilchez247 committed:
Actualizo poetry.lock

Avanza #30

josevilchez247 committed:
Elimino task.py

Avanza #28

josevilchez247 committed:
Actualizo iv.yaml

Avanza #28

josevilchez247 commented:
Ya he vuelto a dejar los ficheros como los tenía en las primeras entregas del objetivo. Cada commit con su issue correspondiente también.

JJ commented:
> Ya he vuelto a dejar los ficheros como los tenía en las primeras entregas del objetivo. Cada commit con su issue correspondiente también.

¿Vas a usar poe en vez de invoke?

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,20 @@
+    poetry config virtualenvs.create false && \
+    poetry install --only-root
+
+ENTRYPOINT ["poetry", "run", "test"]
```
Tienes que usar el gestor de tareas aquí.

josevilchez247 replied:
Cierto lo tenia puesto con poe antes no se si es que ya con tantos cambios que probé deje poetry sin querer lo cambio ahora.

JJ commented on pyproject.toml:
```diff
@@ -5,11 +5,10 @@ description = "Calcula la ruta más óptima para el repostaje de gasolina en un
 
 [tool.poetry.dependencies]
-python = ">=3.7"
+python = ">=3.7,<4.0.0"
```
No existe Python 4

josevilchez247 replied:
Ya, pero uno de los problemas que me dio con respecto a las versiones lo busqué por internet y la solución que daba era poner menor estricto

JJ replied:
¿Qué problema? ¿En qué issue está? ¿Cómo puede solucionar ese problema poner versiones de Python *que no existen*?

josevilchez247 replied:
Al principio yo puse python = ^3.7 y me dio un error, pero si tiene razon lo cambio.

JJ replied:
> Al principio yo puse python = ^3.7 y me dio un error, pero si tiene razon lo cambio.

De veras, tienes que centrarte un poco. Si te dio un error, tendrías que ver el error que te da. Crear un issue para el mismo. Además, no estás cambiando *eso* sino añadiendo un límite a algo que no es lo que comentas.

JJ commented:
Cambiar una dependencia como poetry por poe necesita también su issue que explique el problema; el commit debe referirse a ese issue.

josevilchez247 commented:
Profesor lo de poe e invoke no es que este haciendo ahora un cambio, siempre ha sido poe desde el objetivo 3 donde ya justifique su elección etc, simplemente que en la última prueba ya por probar algo dije voy a probar con invoke y por eso en los ficheros esta el invoke pero fue eso una prueba.

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,20 @@
+
+RUN pip install poetry && \
+    poetry config virtualenvs.create false && \
+    poetry install --only-root
```
¿Por qué haces esto? ¿Qué implica que lo hagas?

JJ replied:
Has resuelto la conversación sin responder a la pregunta. Convendría que lo hicieras, y entendieras por qué lo haces.

JJ commented:
> Profesor lo de poe e invoke no es que este haciendo ahora un cambio, siempre ha sido poe desde el objetivo 3 donde ya justifique su elección etc, simplemente que en la última prueba ya por probar algo dije voy a probar con invoke y por eso en los ficheros esta el invoke pero fue eso una prueba.

Estás haciendo un cambio en la base de código, tiene que estar justificado por un issue y este por una historia de usuario.

josevilchez247 committed:
Actualizo pyproject.toml

#28, error con la versión de python.

JJ commented on pyproject.toml:
```diff
@@ -5,7 +5,7 @@ description = "Calcula la ruta más óptima para el repostaje de gasolina en un
 
 [tool.poetry.dependencies]
-python = ">=3.7,<4.0.0"
+python = "^3.9"
```
Por favor, Jose, céntrate un poco. Para empezar, no puedes poner en un commit cosas obvias como el nombre del fichero que estás cambiando. Di por qué lo haces. Pero, lo más importante, ¡estás cambiando la versión de Python y la forma de expresarla en vez de simplemente borrar versiones inexistentes! ¿Por qué lo haces?
Por favor, tratad de entender lo que hacéis y por qué lo hacéis, y qué implicación tienen los cambios y la forma de documentarlos con mensajes de commit.

josevilchez247 replied:
Cierto simplemente tendría que haber borrado la versión inexistente. Lo cambio

josevilchez247 committed:
Actualizo Dockerfile

Avanza #29

JJ commented on Dockerfile:
```diff
@@ -1,20 +1,30 @@
-RUN groupadd -g 1000 -r usertest && \
-    useradd -u 1000 -m -r -g usertest usertest
+# Crear usuario que ejecutará el contenedor y el grupo al que pertenece
+RUN groupadd -r usertest && useradd -m --no-log-init -r -g usertest usertest
```
Que no es eso, Jose.

JJ commented on Dockerfile:
```diff
@@ -1,20 +1,30 @@
+RUN pip3 install poetry && \
     poetry config virtualenvs.create false && \
-    poetry install --only-root
+    poetry install
```
.... Hum, puede ser esto 🤔
En todo caso, no hagas más cambios que los estrictamente necesarios, porque al final no sabes qué es lo que va a ser y lo que no.

josevilchez247 replied:
vale quito el only-root

josevilchez247 committed:
Actualizo Dockerfile

#29 Avanza

josevilchez247 committed:
Cambio versión de python

Avanza #28, borro versiones inexistentes de Python.

josevilchez247 committed:
Añado las comparaciones de contenedores

#28 justifico la elección del contendor con las comparaciones que hice con otras versiones de python

josevilchez247 committed:
Añado el path al contenedor

#29 avanza

JJ reviewed (changes requested):
Por favor, entiende lo que estás haciendo y los errores que provoca.

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,30 @@
+WORKDIR /app/test
+
+# Mejorar la versión de pip: eliminar warnings
+RUN python -m pip install --upgrade pip
```
¿para qué actualizas pip?

JJ commented on Dockerfile:
```diff
@@ -0,0 +1,30 @@
+USER usertest
+
+# Copiar pyproject.toml y poetry.lock para poder instalar dependencias más tarde
+COPY pyproject.toml poetry.lock ./
```
¿A qué directorio está copiando esto?

josevilchez247 committed:
Modifico Dockerfile problemas de permisos

#30 avanza

josevilchez247 commented:
@JJ No lo consigo en mi maquina no me da problemas ahora vengo aquí y cuando no es un error es otro.

josevilchez247 committed:
Actualizo nuevo pyproject.toml

Arregla problema de versiones #30 avanza

josevilchez247 committed:
Actualizo poetry.lock

Avanza #30
